pkg/errors/error.go

```diff
@@ -34,7 +34,7 @@ func (e *SnapstoreError) Error() string {
 	return e.Message
 }
 
-// AnyError checks whether err is nil or not.
-func AnyError(err error) bool {
+// IsErrNotNil checks whether err is nil or not and return boolean.
+func IsErrNotNil(err error) bool {
```

Comment:
```suggestion
func ErrNotNil(err error) bool {
```
Reads better imo

Also, why are we changing the name of this func in the first place?

Reply:
`IsErrNotNil` making more sense to me as it is checking and returning bool
> Also, why are we changing the name of this func in the first place?

`AnyError` name was not making sense to me.

Reply:
Okay, fair enough
No hard opinion

Reply:
If you look at the method in isolation then your arguments are valid. However always look from the perspective of its usage which will give you the reason this function was introduced in the first place.
```go
if err := retry.OnError(retry.DefaultBackoff, errors.AnyError, func() error {
		return m.RemoveMember(ctx)
	}); err != nil {
		return fmt.Errorf("unable to remove the member %v", err)
	}
```
The second parameter that `OnError` takes is `RetriableError` which is a function which takes an error and returns a boolean indicating if a retry is required. One implementation of `RetriableError` is `AnyError` which always retry irrespective of an error. The correct way to do this would to change the implementation of this method to https://github.com/gardener/dependency-watchdog/blob/master/internal/util/retry.go#L111-L113. So irrespective of the error it will always retry.